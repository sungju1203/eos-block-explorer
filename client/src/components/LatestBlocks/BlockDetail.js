import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

import { fromTheme } from "helper"

const Wrapper = styled.div`
  background: ${fromTheme('primary')};
  color: ${fromTheme('white')};
  font-size: 12px;
  text-align: left;
  word-wrap: break-word;
  padding: 10px;
`

const P = styled.div`
  margin-left: 10px;
`

const BlockDetail = ({
  previous,
  timestamp,
  transaction_mroot,
  action_mroot,
  block_mroot,
  producer,
  schedule_version,
  producer_signature,
  input_transactions,
  id,
  block_num,
  ref_block_prefix,
}) => (
  <Wrapper>
    {"{"}
    <br/>
    <P>previous: {previous},</P>
    <P>timestamp: {timestamp},</P>
    <P>transaction_mroot: {transaction_mroot},</P>
    <P>action_mroot: {action_mroot},</P>
    <P>block_mroot: {block_mroot},</P>
    <P>producer: {producer},</P>
    <P>schedule_version: {schedule_version},</P>
    <P>producer_signature: {producer_signature},</P>
    <P>input_transactions: [ {input_transactions} ],</P>
    <P>id: {id},</P>
    <P>block_num: {block_num},</P>
    <P>ref_block_prefix: {ref_block_prefix},</P>
    {"}"}
  </Wrapper>
)

BlockDetail.propTypes = {
  previous: PropTypes.string.isRequired,
  timestamp: PropTypes.string.isRequired,
  transaction_mroot: PropTypes.string.isRequired,
  action_mroot: PropTypes.string.isRequired,
  block_mroot: PropTypes.string.isRequired,
  producer: PropTypes.string.isRequired,
  schedule_version: PropTypes.number.isRequired,
  producer_signature: PropTypes.string.isRequired,
  input_transactions: PropTypes.array.isRequired,
  id: PropTypes.string.isRequired,
  block_num: PropTypes.number.isRequired,
  ref_block_prefix: PropTypes.number.isRequired,
}

export default BlockDetail