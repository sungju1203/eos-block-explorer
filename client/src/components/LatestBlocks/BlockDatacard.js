import React, { Component } from "react"
import styled from "styled-components"
import moment from "moment"
import PropTypes from "prop-types"

import { fromTheme } from "helper"
import BlockNumber from "./BlockNumber"
import BlockInfo from "./BlockInfo"
import BlockDetail from "./BlockDetail"

const InfoWrapper = styled.div`
  background: ${fromTheme('white')};
  display: flex;
  margin-top: 10px;
  &:hover {
    cursor: pointer;
  }
`

class BlockDatacard extends Component {
  state = { isExpanded: false }

  showDetail = () => {
    this.setState({ isExpanded: !this.state.isExpanded })
  }

  render() {
    const { isExpanded } = this.state
    const { block_num, id, timestamp, producer, input_transactions } = this.props
    return (
      <React.Fragment>
        <InfoWrapper onClick={this.showDetail}>
          <BlockNumber blockNumber={block_num} id={id} />
          <BlockInfo
            id={id}
            timestamp={moment(timestamp).format('YYYY/MM/DD HH:mm:ss')}
            producer={producer}
            transactions={input_transactions.length}
          />
        </InfoWrapper>
        {isExpanded &&
          <BlockDetail {...this.props} />
        }
      </React.Fragment>
    )
  }
}

BlockDatacard.propTypes = {
  block_num: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  timestamp: PropTypes.string.isRequired,
  producer: PropTypes.string.isRequired,
  input_transactions: PropTypes.array.isRequired,
}

InfoWrapper.displayName = 'InfoWrapper'
BlockDetail.displayName = 'BlockDetail'

export default BlockDatacard