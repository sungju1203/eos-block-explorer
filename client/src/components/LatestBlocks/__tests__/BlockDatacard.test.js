import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"

import { fromTheme } from "helper"
import BlockDatacard from "../BlockDatacard"
import BlockDetail from "../BlockDetail"

test('Block Datacard expands', () => {
  const props = {
    previous: '0015ef8791afd56684c335a22703477680973404b27b8a08853bed31c059c40c',
    timestamp: '2018-05-17T04:34:36.500',
    transaction_mroot: '411f319c4b1d68df231455b933d00e9c9231534e6b7d45826ea86d3b4dd12289',
    action_mroot: '9a19514d4a7aa1345bd7ba7f48710734950bcfd7cf15a6261f378841e3cb73bf',
    block_mroot: '09322ac92d6030e017221d74020e5b09f3d53c2d37ab58d9fa429ae5c982c3c3',
    producer: 'eosio',
    schedule_version: 0,
    producer_signature: 'SIG_K1_KAscgo6hWs3R3iyEMNY1dGWzyMbZ49QUk7rybTc4Tzrbrkam3za81i7gMkXuQYXUYNf8Lha5t4pC3f5muG3ocUZqPcJooa',
    input_transactions: [],
    id: '0015ef887c1badff4d207fc03af2ceb78f935ba380e45b71b601901ddb7e508d',
    block_num: 1437576,
    ref_block_prefix: 3229556813,
  }

  const datacard = mount(<BlockDatacard {...props} />)
  const InfoWrapper = datacard.find('InfoWrapper')
  InfoWrapper.instance().showDetail = jest.fn()
  datacard.update()
  // expect(datacard.instance().showDetail).toBeCalledWith()
})