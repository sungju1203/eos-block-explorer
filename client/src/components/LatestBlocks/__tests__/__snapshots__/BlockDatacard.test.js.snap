// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block Datacard expands 1`] = `
<div
  className="sc-dnqmqq gPmenV"
  onClick={[Function]}
>
  <div
    className="sc-bdVaJa jGVUjf"
  >
    <div>
       Block 
      1437576
       
    </div>
    <div
      className="sc-bwzfXH jPuMXn"
    >
       
      0015ef887c1badff4d207fc03af2ceb78f935ba380e45b71b601901ddb7e508d
       
    </div>
  </div>
  <div
    className="sc-EHOje kovVMd"
  >
    <div
      className="sc-bZQynM kzlsZH"
    >
      produced by  
      <div
        className="sc-ifAKCX eXRWbD"
      >
        eosio
      </div>
    </div>
    <div
      className="sc-bZQynM kzlsZH"
    >
      at  
      <div
        className="sc-ifAKCX eXRWbD"
      >
        2018/05/17 04:34:36
      </div>
    </div>
    <div
      className="sc-bZQynM kzlsZH"
    >
      <div
        className="sc-ifAKCX eXRWbD"
      >
        0
      </div>
        transactions
    </div>
  </div>
</div>
`;
